inference-engine/src/inference_engine/include/ie/ie_blob.h

```diff
@@ -187,6 +187,28 @@ class INFERENCE_ENGINE_API_CLASS(Blob) {
      */
     virtual bool deallocate() noexcept = 0;
 
+    /**
+     * @brief Set new shape for blob, deallocate/allocate if new total size is bigger than previous one.
+     *
+     * @param dims new shape
+     */
+    void setShape(const SizeVector& dims) {
+        if (proper_product(dims) > proper_product(getTensorDesc().getDims())) {
+            // New blob shape requires more memory than old one -- reallocate
+            if (!deallocate())
+                IE_THROW() << "Cannot deallocate blob while an attempt to enlarge blob area in setShape.";
+
+            getTensorDesc().setDims(dims);  // Q: If rank is changed, who should set the layout?
+            // A: Old and new ranks should match as well as layouts; won't be a limitation in the new API
```

Comment:
Do we really need such comments?
I propose to add more information in the method description if we need.

Moreover maybe should we need an assert